src/fondant/core/component_spec.py

```diff
@@ -411,7 +411,7 @@ def _validate_mappings(self) -> None:
             for key, value in mapping.items():
                 if not isinstance(value, (str, pa.DataType)):
                     msg = f"Unexpected type {type(value)} received for key {key} in {name} mapping"
-                    raise InvalidPipelineDefinition(msg)
+                    raise InvalidWorkspaceDefinition(msg)
```

Comment:
Is `InvalidWorkspaceDefinition` a logical name here? Shouldn't we use `InvalidOperation` or something similar?